crates/script/src/lib.rs

```diff
@@ -233,6 +233,16 @@ impl ScriptArgs {
             evm_opts.sender = sender;
         }
 
+        // If no sender derived yet, try to derive from account
+        if evm_opts.sender == Address::ZERO {
```

Comment:
while this works, I'd prefer if we check if there is a sender value provided in the ScriptArgs instead

or since this is already done tin the maybe_derive_sender_from_account, we can skip the zero check here and use an else if